app/controllers/organizations/staff/invitations_controller.rb

```diff
@@ -3,29 +3,57 @@ class Organizations::Staff::InvitationsController < Devise::InvitationsControlle
 
   layout "dashboard", only: [:new, :create]
 
+  # This action is for generally creating any User, and it is currently unused.
+  # See the other sub InvitationsControllers for the used `new` actions.
   def new
-    authorize! StaffAccount, context: {organization: Current.organization},
+    authorize! User, context: {organization: Current.organization},
       with: Organizations::InvitationPolicy
 
     @user = User.new
-    @staff = StaffAccount.new(user: @user)
   end
 
   def create
-    authorize! StaffAccount, context: {organization: Current.organization},
-      with: Organizations::InvitationPolicy
+    # We have different return paths based on the roles provided in the params.
+    # If you extend this, make sure new paths have their own authz!
+    case user_params[:roles]
```

Comment:
Just a thought here - it may not work (but I think it should). But, I wonder for each new controller `FostererInvitationsController` and `StaffInvitationsController` if they instead inherited from the `InvitationsController` then we could use the same `#new` logic, and for `#create` we could put most (or all( of this case this logic in each respective controller (or call `super` if it made sense to pass some of the work to the parent action)? It might be a way for us to keep the logic in these controllers rather than have it split between the three..if that makes sense? I don't think there is an issue in inheriting from `InvitationsController` over `Organizations::BaseController`